cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -70,6 +70,20 @@ func ValidateMasterConfiguration(c *kubeadm.MasterConfiguration) field.ErrorList
 	allErrs = append(allErrs, ValidateToken(c.Token, field.NewPath("token"))...)
 	allErrs = append(allErrs, ValidateFeatureGates(c.FeatureGates, field.NewPath("feature-gates"))...)
 	allErrs = append(allErrs, ValidateAPIEndpoint(c, field.NewPath("api-endpoint"))...)
+	allErrs = append(allErrs, ValidateProxy(c.KubeProxy, field.NewPath("kube-proxy"))...)
+	return allErrs
+}
+
+func ValidateProxy(c kubeadm.KubeProxy, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
```

Comment:
You should not have much own logic here, instead this should call `pkg/proxy/apis/kubeproxyconfig/validation.Validate()`

Reply:
ok, the kubeproxyconfig/validation.Validate() takes config *kubeproxyconfig.KubeProxyConfiguration as an arg, but the masterconfig has *kubeproxyconfigv1alpha1.KubeProxyConfiguration. Should I be converting the KubeProxyConfiguration?